lib/inputs/rpm/static.ts

```diff
@@ -14,62 +14,21 @@ export const getRpmDbFileContentAction: ExtractAction = {
 
 export async function getRpmDbFileContent(
   extractedLayers: ExtractedLayers,
-  tmpDirPath?: string,
 ): Promise<string> {
   const rpmDb = getContentAsBuffer(extractedLayers, getRpmDbFileContentAction);
   if (!rpmDb) {
     return "";
   }
 
-  const filePath = generateTempFileName(tmpDirPath);
-  await writeToFile(filePath, rpmDb);
-
   try {
-    // This is the tool that is expected to be found on the system:
-    // https://github.com/snyk/go-rpmdb
-    const cmdOutput = await execute("rpmdb", ["-f", filePath]).catch(
-      handleError,
-    );
-    return cmdOutput.stdout || "";
-  } finally {
-    await removeFile(filePath);
-  }
-}
-
-function handleError(error): CmdOutput | never {
-  const stderr = error.stderr;
-  if (typeof stderr === "string" && stderr.indexOf("not found") >= 0) {
-    return { stdout: "", stderr: "" };
+    const parserResponse = await getPackages(rpmDb);
+    if (parserResponse.error !== undefined) {
+      throw parserResponse.error;
+    }
+    return parserResponse.response;
+  } catch (error) {
+    debug("An error occurred while analysing RPM packages");
+    debug(error);
+    return "";
   }
```

Comment:
Notice that regardless of the error type, we prefer to log for now.
Do you think we should allow the error to propagate up? I think I prefer to fail hard instead of silently returning an empty response as if we didn't process anything...

Reply:
I haven't gone too much into the details here but I propose:
1. in terms of error throwing, make no changes from the existing state
2. in terms of logging and propagating **debug data**, do whatever you think is good for us to improve things

Reply:
I think the old flow basically never threw, we haven't had any cases where it did, hence the current change of not throwing should be OK as well!